builtin/mainmenu/tab_online.lua

```diff
@@ -55,6 +55,43 @@ local function get_sorted_servers()
 	return servers
 end
 
+-- The selected server is persisted in the "address" and "remote_port" settings.
+local function set_selected_server(server)
+	-- reset selection
+	if server == nil then
+		core.settings:set("address", "")
+		core.settings:set("remote_port", "30000")
+		return
```

Comment:
I've switched to `core.settings:remove` because that should be a safe improvement, though note that I was just keeping original code here.

The original code did not reset `gamedata` and I don't see why it would have to. `gamedata.fav` and `gamedata.serverdescription` are only used if there is a selected server as far as I can tell. The whole `gamedata` abuse here feels a bit wrong tbh.

That said I think the code for resetting the selection is essentially dead code anyways, because the only place where it is called is when joining a server, and that sets these settings. So with this PR, there currently is essentially no way (beyond the server ceasing to be in the list) to "deselect" a server, which I think is fine.

I could remove the code for resetting the selection altogether (and maybe try to clean up the rest a bit).
Alternatively I could make `set_selected_server` not persist the selection, but instead use local variables, which could and would be reset independently of the settings, to preserve the old behavior (e.g. restarting Luanti or joining a server loses selection). I'm currently slightly preferring this, because it makes for a more obvious / simpler PR.

Reply:
Implemented the latter approach: https://github.com/minetest/minetest/pull/15466/commits/c926ce72abc51c2c24a9115e2a0116849d9ef797